zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/user/UpdateUserMultiPartitionTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.processing.user;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.tuple;
+
+import io.camunda.zeebe.engine.util.EngineRule;
+import io.camunda.zeebe.protocol.impl.record.value.user.UserRecord;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.RecordType;
+import io.camunda.zeebe.protocol.record.intent.CommandDistributionIntent;
+import io.camunda.zeebe.protocol.record.intent.UserIntent;
+import io.camunda.zeebe.protocol.record.value.CommandDistributionRecordValue;
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestWatcher;
+
+public class UpdateUserMultiPartitionTest {
+  private static final int PARTITION_COUNT = 3;
+
+  @ClassRule public static final EngineRule ENGINE = EngineRule.multiplePartition(PARTITION_COUNT);
+
+  @Rule public final TestWatcher testWatcher = new RecordingExporterTestWatcher();
+
+  @Test
+  public void shouldDistributeUserUpdateCommand() {
+    // when
+    final var username = UUID.randomUUID().toString();
+
+    ENGINE
+        .user()
+        .newUser(username)
+        .withName("Foo Bar")
+        .withEmail("foo@bar.com")
+        .withPassword("password")
+        .create();
+
+    ENGINE
+        .user()
+        .existingUser(
+            new UserRecord()
+                .setUsername(username)
+                .setName("Foo Bar")
+                .setEmail("foo@bar.com")
+                .setPassword("password"))
+        .withUpdatedName("Bar Foo")
+        .withUpdatedEmail("bar@foo.com")
+        .update();
+
+    assertThat(
+            RecordingExporter.records()
+                .withPartitionId(1)
+                .limit(record -> record.getIntent().equals(CommandDistributionIntent.FINISHED)))
```

Comment:
❌ This is the reason the test is flaky. What we say is here is to get all the records up until the first record with the intent `CommandDistributionIntent.FINISHED`. As we create the user before the update, and a user creationg also gets distributed this isn't correct. We must wait until found a second record with this intent. We can easily do this with:

```suggestion
                .limitByCount(record -> record.getIntent().equals(CommandDistributionIntent.FINISHED), 2))
```

After this it may still be flaky. Ideally we wait after the user creation and before the update that the distribution of the create command is finished:

```java
    RecordingExporter.commandDistributionRecords(CommandDistributionIntent.FINISHED)
        .withDistributionIntent(UserIntent.CREATE)
        .await();
```

Reply:
❌ I saw you've added the await, which is good. But you forgot to change the `limit` to `limitByCount`. Without this the test is still flaky 😉